cluster-test/test-orchestrator/test_yaml_parser.go

```diff
@@ -0,0 +1,308 @@
+		panic("empty command")
+	}
+
+	return &Command{
+		Label: label,
+		Cmd:   split[0],
+		Args:  split[1:],
+	}
+}
+
+func parseMeasure(msData []string) []*Measurement {
+	var ms []*Measurement
+	for _, mData := range msData {
+		ms = append(ms, parseMeasurement(mData))
+	}
+	return ms
+}
+
+func parseMeasurement(str string) *Measurement {
+	defer wrapPanicf("in parse measure '%s'", str)
+
+	split := split(str)
+	var measurementArgs = split
+
+	if len(split) < 3 {
+		panic("contains too few fields")
+	}
+
+	// search for assertion
+	var assertion *Assertion
+	for i, s := range split {
+		if s == "is" || s == "in" {
+			interval := strings.Join(split[i+1:], "")
+			assertion = parseAssertion(split[i], interval)
+			measurementArgs = split[3:i]
+		}
+	}
+
+	return &Measurement{
+		Start:     split[0],
+		End:       split[1],
+		Quantity:  split[2],
+		Args:      measurementArgs,
+		Assertion: assertion,
+	}
+}
+
+func parseAssertion(typStr string, arg string) *Assertion {
+	defer wrapPanicf("in parse assertion '%s %s'", typStr, arg)
+
+	switch typStr {
+	case "is":
+		typ := AssertionTypeIs
+		v := parseValue(arg)
+		return &Assertion{
+			Type: typ,
+			V1:   v,
+		}
+
+	case "in":
+		typ := AssertionTypeIn
+		v1, v2 := parseRange(arg)
+		return &Assertion{
+			Type: typ,
+			V1:   v1,
+			V2:   v2,
+		}
+	}
+
+	panic("not valid assertion type")
+}
+
+func parseRange(rng string) (v1, v2 Value) {
+	defer wrapPanicf("in parse range '%s'", rng)
+
+	if rng[0] != '(' || rng[len(rng)-1] != ')' {
+		panic("should be enclosed by parenthesis")
+	}
+	split := strings.Split(rng[1:len(rng)-1], ",")
+	if len(split) != 2 {
+		panic("should be split by a comma")
+	}
+
+	v1 = parseValue(split[0])
+	v2 = parseValue(split[1])
+
+	if reflect.TypeOf(v1) != reflect.TypeOf(v2) {
+		panic(fmt.Sprintf("types %T %T should be equal", v1, v2))
+	}
+
+	return v1, v2
+}
+
+func parseValue(str string) Value {
+	defer wrapPanicf("in parse value '%s", str)
+
+	// First check if the input is a numer or expression.
+	v, err := Eval(str)
+	if err == nil {
+		return v
+	}
+
+	// Then check if the input is a duration. Duration check needs
+	// to be after Eval to prevent "0" to parse as a duration.
+	d, err := time.ParseDuration(str)
+	if err == nil {
+		return Value(d)
+	}
+
+	panic("value is not a number duration or expression")
+}
+
+// split splits a whitespace separted string
+func split(str string) []string {
+	defer wrapPanicf("in split '%s'", str)
+
+	scanner := bufio.NewScanner(strings.NewReader(str))
+	scanner.Split(bufio.ScanWords)
+
+	var wrds []string
+	for scanner.Scan() {
+		wrds = append(wrds, scanner.Text())
+	}
+	if err := scanner.Err(); err != nil {
+		panic(err)
+	}
+
+	return wrds
+}
+
+// replace finds occurrences of varsData and replaces them by the respective
+// element in the runsData.
+func replace(str string, varsData []string, runData []string) string {
+	for i := range varsData {
+		str = strings.Replace(str, varsData[i], runData[i], -1)
```

Comment:
Yeah, adding the check when parsing the vars might be nice. otherwise if I define a var called `i` all i's in all words will be replaced (E.g. a description like "check if ringpop reincarnates nodes" becomes "check 5f r5ngpop re5ncarnates nodes" when i=5).